onnxruntime/core/graph/contrib_ops/bert_defs.cc

```diff
@@ -1215,7 +1215,23 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
         }));
 
 constexpr const char* GemmaRotaryEmbedding_ver1_doc = R"DOC(
-GemmaRotaryEmbedding is the implementation of rotary positional embeddings (RoPE). It includes sin and cos in GemmaRotaryEmbedding forward and apply_rotary_pos_emb
+GemmaRotaryEmbedding is the implementation of below part of rotary positional embeddings (RoPE). It implements below from modeling_gemma.py.
+- Only support fp16, other types are not support
+- rotate_half is not included
+
+class LlamaRotaryEmbedding(nn.Module):
```

Comment:
This doesn't explain the math. For example, what are inputs/outputs of this op? Are they inputs/outputs of `LlamaRotaryEmbedding`? Or are they inputs/outputs of `apply_rotary_pos_emb`? Do you write two fused kernels, one for `LlamaRotaryEmbedding` and the other one for `apply_rotary_pos_emb`? Let's consider fusion of MatMul and Add as an example. I'd using onnxscript syntax to define the fused sub-graph in Python:
```
# The computation graph below (defined by onnxscript) is the math representation of this kernel.
# This kernel has 3 inputs, `x`, `y`, and `bias`. This kernel produces only one output tensor.
onnxscript.script()
def matmul_add_fusion(x, y, bias):
  z = op.MatMul(x, y)
  w = op.Add(z, bias)
  return w
```

Reply:
I see what you mean. Fied